src/state.rs

```diff
@@ -60,6 +63,25 @@ impl NetworkState {
         }
     }
 
+    /// Persists the contracts stored on the [`NetworkState`] specifying a
+    /// path where they should be stored in the disk.
+    pub fn persist(
+        &self,
+        ctor: fn() -> Result<DiskBackend, PersistError>,
+    ) -> Result<PersistedId, PersistError> {
+        Persistance::persist(&BackendCtor::new(ctor), &self.contracts)
+    }
+
+    /// Given a [`PersistedId`] restores the [`Hamt`] which stores the contracts
+    /// of the entire blockchain state.
+    pub fn restore(&self, id: PersistedId) -> Result<Self, PersistError> {
+        Ok(NetworkState {
+            block_height: self.block_height,
+            contracts: Hamt::from_generic(&id.restore()?)?,
+            modules: self.modules.clone(),
+        })
+    }
```

Comment:
The idea of this API was to have a mutable self and just modify `self.contracts`; since otherwise you're creating the instance twice.
Is there a reason why you opted for create a new instance instead?

Reply:
It doesn't make sense to use `restore` requiring a mut ref to a `NetworkState` since that means that we should already have an instance which simply does not make sense in some cases.

Reply:
`restore` is an instance method (it has `self`) so you will *always* have an instance.

Reply:
Agreed